consoleme/handlers/base.py

```diff
@@ -113,6 +113,13 @@ def get_current_user(self):
 class BaseHandler(tornado.web.RequestHandler):
     """Default BaseHandler."""
 
+    def get_request_ip(self):
+        if config.get("auth.remote_ip.use_xforwarded_for"):
+            return self.request.headers.get(
+                "X-Forwarded-For", self.request.remote_ip
+            ).split(",")[0]
+        return self.request.remote_ip
```

Comment:
Is this provided by Tornado?

Reply:
Yes. Normally it would present the request IP (The one closest to ConsoleMe, which, when behind a load balancer, would  be a load balancer IP. Not very valuable). If not behind LB, it would be the client IP.

X-Forwarded-For can be forged by the client when ConsoleMe is not behind a load balancer, so there's a risk.

Reply:
I can be wrong here but I believe even if ConsoleMe was behind a load balancer there is risk. This risk is dependent on how the load balancer treats X-Forwarded-For. For example, if a proxy pushes a request to the load balancer, then the load balancer _may_ trust X-Forwarded-For. It entirely depends on the load balancer's settings and configuration. When you add a setup where there are multiple proxies, you increase not only the complexity but risk that X-Forwarded-For can be forged by a malicious user.

Reply:
This risk is why I suggested we require an explicit configuration, rather than the previous behavior of "trust the header if present, if not fall through to the IP". This way, when a user is configuring the app, they can say "I know that my load balancer always sets `X-Forwarded-For` correctly, and trust that value. If it is missing, then the request will fail because there's a problem with the underlying configuration. If the load balancer sets a different header, then the user can configure consoleme to trust just that header instead.

In either case, it is still important that the user properly configures both consoleme (to trust the appropriate header) as well as the load balancer (to always correctly set a specific header, and not trust a client pretending to be a proxy). Properly configuring the load balancer or whatever else is in front of consoleme is outside the scope of what consoleme can detect or be responsible for. This setting just makes the decision explicit.

Reply:
Yep! Makes total sense to me. 👍